### PR TITLE
Start architecture documentation

### PR DESCRIPTION
**Pull request by anandthakker**
Occurred to me as I was reading through / learning how the rendering subsystem works in more detail that capturing the notes might be a good way to start an architecture doc.  (This is something I've long wanted to add, as a way to make contributing to GL JS more accessible.)

### CODE REVIEW

**Comment by mollymerp** [comment]
yessss thank you so much for taking this on!!

**Comment by andrewharvey** [comment]
Thank you, this would be a big help!

**Comment by anandthakker** [comment]
I had a few questions while reading through `SymbolBucket`, and added them as TODO's here so as to remember to get & document answers to them before merging.  @lucaswoj @jfirebaugh @mourner could one of you take a 👀 ?

File: src/data/bucket/circle_bucket.js [hunk 1]
```diff
@@ -81,4 +81,6 @@ class CircleBucket extends Bucket {
     }
 }
 
+CircleBucket.programInterface = circleInterface;
```
Comment (lucaswoj) [inline src/data/bucket/circle_bucket.js]: I'm confused -- is this related to the architectural documentation or something else?

Comment (lucaswoj) [inline src/data/bucket/circle_bucket.js]: Ah. This is for the `describe_shaders` utility 👍

**Pull request merged**